lib/stdlib/doc/src/re.xml

```diff
@@ -2478,6 +2477,11 @@ foo\Kbar</code>
       <tag>xdigit</tag><item>Hexadecimal digits</item>
     </taglist>
 
+    <p>There is another character class, <c>ascii</c>, that erroneously matches
+      Latin-1 characters instead of the 0-127 range specified by POSIX. This
+      cannot be fixed without altering the behaviour of other classes, so we
+      recommend matching the range with <c>[\x01-\x7f]</c> instead.</p>
```

Comment:
If it's not too much trouble to explain further ... for my own knowledge is this related to locale and the underlying regex library or something else?

Reply:
it's a bit of both. The underlying PCRE lib ties the ascii range with 'printable' characters range, which is only incidentally ascii if you're using the C locale. For historical reasons, Erlang sets the locale default to Latin-1 which extends the printable range and accidentally extends the ascii range at the same time.